In [2]:
%reload_ext sql
%sql sqlite:///./Videogames.db

In [3]:
%sql  select * FROM "vgsales" Limit 100

 * sqlite:///./Videogames.db
Done.


[(1, 'Wii Sports', 'Wii', 2006, 'Sports', 'Nintendo', 41.49, 29.02, 3.77, 8.46, 82.74),
 (2, 'Super Mario Bros.', 'NES', 1985, 'Platform', 'Nintendo', 29.08, 3.58, 6.81, 0.77, 40.24),
 (3, 'Mario Kart Wii', 'Wii', 2008, 'Racing', 'Nintendo', 15.85, 12.88, 3.79, 3.31, 35.82),
 (4, 'Wii Sports Resort', 'Wii', 2009, 'Sports', 'Nintendo', 15.75, 11.01, 3.28, 2.96, 33.0),
 (5, 'Pokemon Red/Pokemon Blue', 'GB', 1996, 'Role-Playing', 'Nintendo', 11.27, 8.89, 10.22, 1.0, 31.37),
 (6, 'Tetris', 'GB', 1989, 'Puzzle', 'Nintendo', 23.2, 2.26, 4.22, 0.58, 30.26),
 (7, 'New Super Mario Bros.', 'DS', 2006, 'Platform', 'Nintendo', 11.38, 9.23, 6.5, 2.9, 30.01),
 (8, 'Wii Play', 'Wii', 2006, 'Misc', 'Nintendo', 14.03, 9.2, 2.93, 2.85, 29.02),
 (9, 'New Super Mario Bros. Wii', 'Wii', 2009, 'Platform', 'Nintendo', 14.59, 7.06, 4.7, 2.26, 28.62),
 (10, 'Duck Hunt', 'NES', 1984, 'Shooter', 'Nintendo', 26.93, 0.63, 0.28, 0.47, 28.31),
 (11, 'Nintendogs', 'DS', 2005, 'Simulation', 'Nintendo', 9.07, 11.0, 1.93, 2.75, 24.76),
 (12, 'Mario Kart DS', 'DS', 2005, 'Racing', 'Nintendo', 9.81, 7.57, 4.13, 1.92, 23.42),
 (13, 'Pokemon Gold/Pokemon Silver', 'GB', 1999, 'Role-Playing', 'Nintendo', 9.0, 6.18, 7.2, 0.71, 23.1),
 (14, 'Wii Fit', 'Wii', 2007, 'Sports', 'Nintendo', 8.94, 8.03, 3.6, 2.15, 22.72),
 (15, 'Wii Fit Plus', 'Wii', 2009, 'Sports', 'Nintendo', 9.09, 8.59, 2.53, 1.79, 22.0),
 (16, 'Kinect Adventures!', 'X360', 2010, 'Misc', 'Microsoft Game Studios', 14.97, 4.94, 0.24, 1.67, 21.82),
 (17, 'Grand Theft Auto V', 'PS3', 2013, 'Action', 'Take-Two Interactive', 7.01, 9.27, 0.97, 4.14, 21.4),
 (18, 'Grand Theft Auto: San Andreas', 'PS2', 2004, 'Action', 'Take-Two Interactive', 9.43, 0.4, 0.41, 10.57, 20.81),
 (19, 'Super Mario World', 'SNES', 1990, 'Platform', 'Nintendo', 12.78, 3.75, 3.54, 0.55, 20.61),
 (20, 'Brain Age: Train Your Brain in Minutes a Day', 'DS', 2005, 'Misc', 'Nintendo', 4.75, 9.26, 4.16, 2.05, 20.22),
 (21, 'Pokemon Diamond/Pokemon Pearl', 'DS', 2006, 'Role-Playing', 'Nintendo', 6.42, 4.52, 6.04, 1.37, 18.36),
 (22, 'Super Mario Land', 'GB', 1989, 'Platform', 'Nintendo', 10.83, 2.71, 4.18, 0.42, 18.14),
 (23, 'Super Mario Bros. 3', 'NES', 1988, 'Platform', 'Nintendo', 9.54, 3.44, 3.84, 0.46, 17.28),
 (24, 'Grand Theft Auto V', 'X360', 2013, 'Action', 'Take-Two Interactive', 9.63, 5.31, 0.06, 1.38, 16.38),
 (25, 'Grand Theft Auto: Vice City', 'PS2', 2002, 'Action', 'Take-Two Interactive', 8.41, 5.49, 0.47, 1.78, 16.15),
 (26, 'Pokemon Ruby/Pokemon Sapphire', 'GBA', 2002, 'Role-Playing', 'Nintendo', 6.06, 3.9, 5.38, 0.5, 15.85),
 (27, 'Pokemon Black/Pokemon White', 'DS', 2010, 'Role-Playing', 'Nintendo', 5.57, 3.28, 5.65, 0.82, 15.32),
 (28, 'Brain Age 2: More Training in Minutes a Day', 'DS', 2005, 'Puzzle', 'Nintendo', 3.44, 5.36, 5.32, 1.18, 15.3),
 (29, 'Gran Turismo 3: A-Spec', 'PS2', 2001, 'Racing', 'Sony Computer Entertainment', 6.85, 5.09, 1.87, 1.16, 14.98),
 (30, 'Call of Duty: Modern Warfare 3', 'X360', 2011, 'Shooter', 'Activision', 9.03, 4.28, 0.13, 1.32, 14.76),
 (31, 'Pokémon Yellow: Special Pikachu Edition', 'GB', 1998, 'Role-Playing', 'Nintendo', 5.89, 5.04, 3.12, 0.59, 14.64),
 (32, 'Call of Duty: Black Ops', 'X360', 2010, 'Shooter', 'Activision', 9.67, 3.73, 0.11, 1.13, 14.64),
 (34, 'Call of Duty: Black Ops 3', 'PS4', 2015, 'Shooter', 'Activision', 5.77, 5.81, 0.35, 2.31, 14.24),
 (35, 'Call of Duty: Black Ops II', 'PS3', 2012, 'Shooter', 'Activision', 4.99, 5.88, 0.65, 2.52, 14.03),
 (36, 'Call of Duty: Black Ops II', 'X360', 2012, 'Shooter', 'Activision', 8.25, 4.3, 0.07, 1.12, 13.73),
 (37, 'Call of Duty: Modern Warfare 2', 'X360', 2009, 'Shooter', 'Activision', 8.52, 3.63, 0.08, 1.29, 13.51),
 (38, 'Call of Duty: Modern Warfare 3', 'PS3', 2011, 'Shooter', 'Activision', 5.54, 5.82, 0.49, 1.62, 13.46),
 (39, 'Grand Theft Auto III', 'PS2', 2001, 'Action', 'Take-Two Interactive', 6.99, 4.51, 0.3, 1.3, 13.1),
 (40, 'Super Smash Bros. Brawl', 'Wii', 2008, 'Fighting', 'Nintendo', 6.75, 2.61, 2.66, 1.02, 13.04),
 (

This article analyzes the data of video games with sales of more than 100,000 copies from 1980 to
2020. The tool used is Navicat Premium--MySql. Display an overview of the data and observe the
relationship between variables such as game types, distribution, publishers, platforms, and regional sales.

Data source: Kaggle "Video Games Sold Over 100,000 Copies", crawled by Python's BeautifulSoup.
Address: https://www.kaggle.com/gregorut/
videogamesales

This analysis is carried out according to
the following steps: Observing dataÿAsk questionsÿCleaning and analyzing dataÿFeelings

1. Observation data

According to the information provided by Kaggle on the data page, the data table has a total of 16,598 records,
recording information about video games with a total sales volume of more than 100,000 copies. The specific fields are as follows:

Rank
Name
Platform
Year
Genre
Publisher
NA_Sales
EU_Sales
JP_Sales
Other_Sales
Global_Sales

2. Asking questions

Preliminary conception based on the above conditions: year, game type, distribution platform, publisher
and the relationship between them and sales are worth exploring, which can be used to analyze which
game is the most popular, which platform, distribution The best business performance, how people's
demand for games changes over time, and the special preferences of consumers in each region for
video games, etc., provide reference and forecast basis for the video game industry.

Questions 
1. What are the number and sales of popular games each year? What is the trend of change?
2. Who are the top 10 companies in terms of total sales of all products?
3. How many types of games are there, how many are the distributions of each type, and what is the total
sales? 
4. For each game console platform, which type of game is the most released? Does each platform have
its own
preferences?
5. What is the total sales volume of each region and the percentage of the global sales? How
has the proportion of sales by
region changed over time? 
6. Which type of game has the highest sales in North America? Which
platform has the highest sales? 
7. Which type of game has the highest sales in Europe? Which platform
has the highest sales?
8. Which type of game has the highest sales in Japan? Which platform has the highest sales?

3. Processing data 

After determining the problem, write the corresponding query command in Navicat according to each problem, and get
the result.

0. Data cleaning

In [4]:
%sql select * from vgsales where year ='n/a' or platform='n/a' or genre= 'n/a' or publisher='n/a' 

 * sqlite:///./Videogames.db
Done.


[]

According to the results, 307 pieces of data are missing, accounting for 1.84% of the total 16,598 pieces of data.Deletion does not affect analysis, perform deletion.


In [5]:
%sql delete from vgsales where year='n/a' or platform or genre = 'n/a' or publisher='n/a'

 * sqlite:///./Videogames.db
0 rows affected.


[]

0.2 Query duplicate values

Guess that the game name is a specific parameter. At the beginning, I thought that the game name is repeated, and the
data is repeated, so query it

In [6]:
%sql select name, count(name) as number from vgsales group by name order by number desc 

 * sqlite:///./Videogames.db
Done.


[('Need for Speed: Most Wanted', 12),
 ('Ratatouille', 9),
 ('Madden NFL 07', 9),
 ('Monopoly', 8),
 ('Madden NFL 08', 8),
 ('LEGO Star Wars II: The Original Trilogy', 8),
 ('LEGO Marvel Super Heroes', 8),
 ('FIFA 14', 8),
 ('Cars', 8),
 ('Wall-E', 7),
 ('Tomb Raider: Legend', 7),
 ('The Sims 2', 7),
 ('The LEGO Movie Videogame', 7),
 ('Terraria', 7),
 ("Surf's Up", 7),
 ('Star Wars The Clone Wars: Republic Heroes', 7),
 ('Spider-Man 3', 7),
 ('Need for Speed: ProStreet', 7),
 ('Need for Speed Carbon', 7),
 ('Need For Speed: Undercover', 7),
 ('NBA Jam', 7),
 ('Major League Baseball 2K11', 7),
 ('Madden NFL 06', 7),
 ('Lego Batman 3: Beyond Gotham', 7),
 ('LEGO The Hobbit', 7),
 ('LEGO Jurassic World', 7),
 ('LEGO Indiana Jones: The Original Adventures', 7),
 ('LEGO Harry Potter: Years 5-7', 7),
 ('LEGO Batman: The Videogame', 7),
 ("Harry Potter and the Sorcerer's Stone", 7),
 ('Harry Potter and the Order of the Phoenix', 7),
 ('Harry Potter and the Half-Blood Prince', 7),
 ('FIFA Soccer 13', 7),
 ('FIFA Soccer 11', 7),
 ('FIFA Soccer 10', 7),
 ('FIFA Soccer 08', 7),
 ('FIFA 15', 7),
 ('Angry Birds Star Wars', 7),
 ('pro evolution soccer 2011', 6),
 ("You Don't Know Jack", 6),
 ('X-Men: The Official Game', 6),
 ('Watch Dogs', 6),
 ('Warriors Orochi 3', 6),
 ('WWE SmackDown vs. Raw 2010', 6),
 ('WWE SmackDown vs. Raw 2009', 6),
 ('WWE SmackDown vs Raw 2008', 6),
 ('Up', 6),
 ('Transformers: Revenge of the Fallen (XBox 360, PS3, & PC Versions)', 6),
 ("Tony Hawk's Pro Skater 3", 6),
 ('Tomb Raider: Underworld', 6),
 ('Tiger Woods PGA Tour 08', 6),
 ('Tiger Woods PGA Tour 07', 6),
 ('The Walking Dead: Season One', 6),
 ('The Sims 2: Pets', 6),
 ('The Simpsons Game', 6),
 ('The Incredibles: Rise of the Underminer', 6),
 ('The Incredible Hulk', 6),
 ('The Golden Compass', 6),
 ('The BIGS 2', 6),
 ('The Amazing Spider-Man 2 (2014)', 6),
 ('Teenage Mutant Ninja Turtles', 6),
 ('Star Wars: The Force Unleashed', 6),
 ('Spider-Man: Web of Shadows', 6),
 ('Spider-Man: Friend or Foe', 6),
 ('Spider-Man 2', 6),
 ('Skylanders: Trap Team', 6),
 ('Skylanders: SuperChargers', 6),
 ('Skylanders SWAP Force', 6),
 ('Rugby World Cup 2015', 6),
 ('Rayman Legends', 6),
 ('Rapala Pro Bass Fishing 2010', 6),
 ('Pro Evolution Soccer 2012', 6),
 ('Pro Evolution Soccer 2010', 6),
 ('Pro Evolution Soccer 2008', 6),
 ('Prince of Persia: The Forgotten Sands', 6),
 ("Pirates of the Caribbean: At World's End", 6),
 ("Peter Jackson's King Kong: The Official Game of the Movie", 6),
 ('Open Season', 6),
 ('Need for Speed Underground 2', 6),
 ('NBA 2K11', 6),
 ('Monsters vs. Aliens', 6),
 ('Minecraft', 6),
 ('Michael Jackson: The Experience', 6),
 ('Marvel: Ultimate Alliance 2', 6),
 ('Marvel: Ultimate Alliance', 6),
 ('Major League Baseball 2K12', 6),
 ('Major League Baseball 2K10', 6),
 ('Madden NFL 2002', 6),
 ('Madden NFL 13', 6),
 ('Madden NFL 09', 6),
 ('Madagascar: Escape 2 Africa', 6),
 ('MX vs. ATV Untamed', 6),
 ('Lego Star Wars: The Force Awakens', 6),
 ('LEGO The Lord of the Rings', 6),
 ('LEGO Star Wars III: The Clone Wars', 6),
 ('LEGO Pirates of the Caribbean: The Video Game', 6),
 ("LEGO Marvel's Avengers", 6),
 ('LEGO Indiana Jones 2: The Adventure Continues', 6),
 ('LEGO Harry Potter: Years 1-4', 6),
 ('LEGO Batman 2: DC Super Heroes', 6),
 ('Kung Fu Panda', 6),
 ('Just Dance 2016', 6),
 ('Just Dance 2015', 6),
 ('Just Dance 2014', 6),
 ('Juiced 2: Hot Import Nights', 6),
 ('Iron Man', 6),
 ('Ice Age: Dawn of the Dinosaurs', 6),
 ('Ice Age 2: The Meltdown', 6),
 ('Harry Potter and the Goblet of Fire', 6),
 ('Harry Potter and the Chamber of Secrets', 6),
 ('Ghostbusters: The Video Game', 6),
 ('G.I. Joe: The Rise of Cobra', 6),
 ('G-Force', 6),
 ('FIFA World Cup Germany 2006', 6),
 ('FIFA Soccer 09', 6),
 ('FIFA Soccer 07', 6),
 ('FIFA 12', 6),
 ('Disney Infinity 2.0: Marvel Super Heroes', 6),
 ('DiRT 2', 6),
 ('Darksiders II', 6),
 ('Cars: Race-O-Rama', 6),
 ('Call of Duty: Ghosts', 6),
 ('Ben 10 Ultimate Alien: Cosmic Destruction', 6),
 ('Avat

There are many games with duplicate values, but at this moment, I thought, maybe it is because the same game
has different versions, released in different years, platforms, and publishers, so it is not counted as duplicate data.


Query the most repeated games in the above results "Need for Speed: Most Wanted" to verify.


In [7]:
%sql select year, publisher, platform from vgsales where name='Need for Speed: Most Wanted'

 * sqlite:///./Videogames.db
Done.


[(2005, 'Electronic Arts', 'PS2'),
 (2012, 'Electronic Arts', 'PS3'),
 (2012, 'Electronic Arts', 'X360'),
 (2005, 'Electronic Arts', 'X360'),
 (2012, 'Electronic Arts', 'PSV'),
 (2005, 'Electronic Arts', 'XB'),
 (2005, 'Electronic Arts', 'GC'),
 (2005, 'Electronic Arts', 'PC'),
 (2013, 'Electronic Arts', 'WiiU'),
 (2005, 'Electronic Arts', 'DS'),
 (2005, 'Electronic Arts', 'GBA'),
 (2012, 'Electronic Arts', 'PC')]

Each piece of data is different, there are differences in year, publisher, and distribution platform,
and all data is kept. After the data has been preliminarily cleaned, start to solve the problems raised above.

1. How many games are released and sold each year? What is the trend?

In [8]:
%sql select year,count(name), round(sum(global_sales),2) from vgsales group by year order by year

 * sqlite:///./Videogames.db
Done.


[(1983, 6, 10.96),
 (1984, 13, 50.09),
 (1985, 13, 53.49),
 (1986, 19, 36.41),
 (1987, 10, 19.76),
 (1988, 13, 46.47),
 (1989, 15, 72.83),
 (1990, 16, 49.39),
 (1991, 41, 32.23),
 (1992, 43, 76.16),
 (1993, 60, 45.98),
 (1994, 120, 79.15),
 (1995, 217, 88.03),
 (1996, 263, 199.15),
 (1997, 289, 200.98),
 (1998, 379, 256.47),
 (1999, 338, 251.27),
 (2000, 349, 201.56),
 (2001, 482, 331.47),
 (2002, 829, 395.52),
 (2003, 775, 357.85),
 (2004, 763, 419.31),
 (2005, 941, 459.94),
 (2006, 1008, 521.04),
 (2007, 1202, 611.13),
 (2008, 1428, 678.9),
 (2009, 1431, 667.3),
 (2010, 1259, 600.45),
 (2011, 1023, 453.46),
 (2012, 564, 312.4),
 (2013, 455, 312.23),
 (2014, 503, 293.91),
 (2015, 528, 237.45),
 (2016, 309, 64.33),
 ('N/A', 245, 88.39)]

The query found that the data recorded video game information from 1980 to 2020. The number of
games with more than 100,000 copies and the total sales showed a trend of first increasing and
then decreasing. The number of games reached the maximum in 2009, which
was 1431 , sales peaked in 2008 at $678,900,000.

At the same time, it was also found that the data of 2019 and 2019 were missing, and the data of 2017 and 2020
were abnormal , so the outliers were deleted.

In [9]:
%sql delete from vgsales where year = 2017 or year=2020

 * sqlite:///./Videogames.db
0 rows affected.


[]

2. Who are the top 10 companies in terms of total sales?
2.1 Ranking according to the number of game releases

In [10]:
%sql select publisher, count(name) as number from vgsales group by publisher order by number desc limit 10

 * sqlite:///./Videogames.db
Done.


[('Electronic Arts', 1344),
 ('Activision', 920),
 ('Ubisoft', 899),
 ('Namco Bandai Games', 845),
 ('Konami Digital Entertainment', 819),
 ('THQ', 712),
 ('Sony Computer Entertainment', 683),
 ('Nintendo', 621),
 ('Sega', 615),
 ('Take-Two Interactive', 411)]

2.2 Ranking based on global sales 

In [11]:
%sql select publisher, count(global_sales) as total_sales from vgsales group by publisher order by total_sales DESC limit 10

 * sqlite:///./Videogames.db
Done.


[('Electronic Arts', 1344),
 ('Activision', 920),
 ('Ubisoft', 899),
 ('Namco Bandai Games', 845),
 ('Konami Digital Entertainment', 819),
 ('THQ', 712),
 ('Sony Computer Entertainment', 683),
 ('Nintendo', 621),
 ('Sega', 615),
 ('Take-Two Interactive', 411)]

According to the above two queries, the list of the top 10 companies is the same, and the ranking
is also the same. The top three are Electronic Art, Activision, Namco Bandai Games, and
Electronic Arts is far behind in terms of the number of shortlisted games and sales. more than other companies.

3. How many types of games are there, how many are the distributions of each type, and what is the total sales?

In [12]:
%sql select genre, count(name) as number, round(sum(global_sales),2) as sales from vgsales group by genre order by number desc

 * sqlite:///./Videogames.db
Done.


[('Action', 3072, 1664.81),
 ('Sports', 2308, 1321.3),
 ('Misc', 1681, 795.9),
 ('Role-Playing', 1400, 851.59),
 ('Shooter', 1279, 1009.6),
 ('Adventure', 1246, 232.47),
 ('Racing', 1232, 714.64),
 ('Platform', 849, 785.87),
 ('Simulation', 834, 364.36),
 ('Fighting', 832, 437.21),
 ('Strategy', 666, 173.03),
 ('Puzzle', 550, 224.68)]

There are 12 categories of games, the most popular in descending order are action, sports,
comprehensive, role-playing, shooting, adventure...

Among them, although the number of games released in the adventure category is large, the sales volume is relatively low compared to other categories
with the same number.

4. Which type of game is the most released on each game platform? Does each platform have its own
preferences?

In [13]:
%sql select platform, genre, count(genre) as number from vgsales group by platform, genre order by platform, number desc

 * sqlite:///./Videogames.db
Done.


[('DC', 'Fighting', 12),
 ('DC', 'Adventure', 11),
 ('DC', 'Sports', 10),
 ('DC', 'Racing', 6),
 ('DC', 'Role-Playing', 4),
 ('DC', 'Shooter', 3),
 ('DC', 'Action', 3),
 ('DC', 'Platform', 2),
 ('DC', 'Simulation', 1),
 ('DS', 'Misc', 393),
 ('DS', 'Action', 343),
 ('DS', 'Simulation', 284),
 ('DS', 'Adventure', 240),
 ('DS', 'Puzzle', 238),
 ('DS', 'Role-Playing', 200),
 ('DS', 'Sports', 148),
 ('DS', 'Platform', 92),
 ('DS', 'Strategy', 79),
 ('DS', 'Racing', 67),
 ('DS', 'Shooter', 42),
 ('DS', 'Fighting', 36),
 ('GB', 'Role-Playing', 21),
 ('GB', 'Platform', 19),
 ('GB', 'Puzzle', 15),
 ('GB', 'Sports', 9),
 ('GB', 'Misc', 8),
 ('GB', 'Strategy', 7),
 ('GB', 'Action', 6),
 ('GB', 'Simulation', 5),
 ('GB', 'Adventure', 5),
 ('GB', 'Racing', 2),
 ('GB', 'Shooter', 1),
 ('GBA', 'Action', 167),
 ('GBA', 'Platform', 142),
 ('GBA', 'Misc', 110),
 ('GBA', 'Sports', 88),
 ('GBA', 'Role-Playing', 73),
 ('GBA', 'Racing', 64),
 ('GBA', 'Puzzle', 41),
 ('GBA', 'Shooter', 40),
 ('GBA', 'Adventure', 38),
 ('GBA', 'Fighting', 23),
 ('GBA', 'Strategy', 18),
 ('GBA', 'Simulation', 18),
 ('GC', 'Sports', 110),
 ('GC', 'Action', 101),
 ('GC', 'Platform', 73),
 ('GC', 'Racing', 63),
 ('GC', 'Shooter', 48),
 ('GC', 'Fighting', 42),
 ('GC', 'Misc', 36),
 ('GC', 'Role-Playing', 27),
 ('GC', 'Adventure', 20),
 ('GC', 'Puzzle', 13),
 ('GC', 'Simulation', 12),
 ('GC', 'Strategy', 11),
 ('GEN', 'Platform', 7),
 ('GEN', 'Fighting', 5),
 ('GEN', 'Sports', 3),
 ('GEN', 'Role-Playing', 3),
 ('GEN', 'Action', 3),
 ('GEN', 'Adventure', 2),
 ('GEN', 'Strategy', 1),
 ('GEN', 'Shooter', 1),
 ('GEN', 'Racing', 1),
 ('GEN', 'Misc', 1),
 ('GG', 'Platform', 1),
 ('N64', 'Sports', 80),
 ('N64', 'Racing', 57),
 ('N64', 'Action', 38),
 ('N64', 'Platform', 30),
 ('N64', 'Fighting', 29),
 ('N64', 'Shooter', 24),
 ('N64', 'Misc', 18),
 ('N64', 'Puzzle', 12),
 ('N64', 'Simulation', 10),
 ('N64', 'Strategy', 9),
 ('N64', 'Role-Playing', 8),
 ('N64', 'Adventure', 4),
 ('NES', 'Platform', 28),
 ('NES', 'Sports', 14),
 ('NES', 'Puzzle', 14),
 ('NES', 'Action', 13),
 ('NES', 'Role-Playing', 11),
 ('NES', 'Shooter', 7),
 ('NES', 'Racing', 4),
 ('NES', 'Fighting', 4),
 ('NES', 'Misc', 2),
 ('NES', 'Adventure', 1),
 ('NG', 'Fighting', 11),
 ('NG', 'Sports', 1),
 ('PC', 'Strategy', 188),
 ('PC', 'Action', 165),
 ('PC', 'Shooter', 148),
 ('PC', 'Simulation', 115),
 ('PC', 'Role-Playing', 104),
 ('PC', 'Adventure', 65),
 ('PC', 'Racing', 60),
 ('PC', 'Sports', 49),
 ('PC', 'Puzzle', 25),
 ('PC', 'Misc', 24),
 ('PC', 'Platform', 11),
 ('PC', 'Fighting', 6),
 ('PCFX', 'Role-Playing', 1),
 ('PS', 'Sports', 222),
 ('PS', 'Action', 157),
 ('PS', 'Racing', 145),
 ('PS', 'Fighting', 108),
 ('PS', 'Role-Playing', 97),
 ('PS', 'Shooter', 96),
 ('PS', 'Misc', 76),
 ('PS', 'Strategy', 70),
 ('PS', 'Adventure', 69),
 ('PS', 'Platform', 64),
 ('PS', 'Simulation', 60),
 ('PS', 'Puzzle', 32),
 ('PS2', 'Sports', 400),
 ('PS2', 'Action', 348),
 ('PS2', 'Misc', 222),
 ('PS2', 'Racing', 216),
 ('PS2', 'Adventure', 196),
 ('PS2', 'Role-Playing', 187),
 ('PS2', 'Shooter', 160),
 ('PS2', 'Fighting', 150),
 ('PS2', 'Platform', 103),
 ('PS2', 'Simulation', 90),
 ('PS2', 'Strategy', 71),
 ('PS2', 'Puzzle', 18),
 ('PS3', 'Action', 380),
 ('PS3', 'Sports', 213),
 ('PS3', 'Shooter', 156),
 ('PS3', 'Misc', 124),
 ('PS3', 'Role-Playing', 119),
 ('PS3', 'Racing', 92),
 ('PS3', 'Fighting', 76),
 ('PS3', 'Adventure', 74),
 ('PS3', 'Platform', 37),
 ('PS3', 'Simulation', 31),
 ('PS3', 'Strategy', 24),
 ('PS3', 'Puzzle', 3),
 ('PS4', 'Action', 122),
 ('PS4', 'Role-Playing', 46),
 ('PS4', 'Sports', 43),
 ('PS4', 'Shooter', 34),
 ('PS4', 'Adventure', 19),
 ('PS4', 'Racing', 17),
 ('PS4', 'Fighting', 17),
 ('PS4', 'Misc', 15),
 ('PS4', 'Platform', 11),
 ('PS4', 'Strategy', 5),
 ('PS4', 'Simulation', 5),
 ('PS4', 'Puzzle', 1),
 ('PSP', 'Action', 222),
 ('PSP', 'Adventure', 213),
 ('PSP', 'Role-Playing', 192),
 ('PSP', 'Sports', 135),
 ('PSP', 'Misc', 106),
 ('PSP', 'Fighting', 74),
 ('PSP', 'Racing', 65),
 ('PSP', 'St

It is observed that different platforms have different styles. The three most produced games of some
companies are not in the top 50% of the results of query 3. For example, 3DO launched and shortlisted
simulation, puzzle and adventure. The most online GB games are role-playing, platform, and puzzle-solving.

It can be inferred that different platforms have different characteristics. Even if you can’t get a share of the
most popular fields such as action and sports, you can still achieve good results on the platform that is suitable and good at.

5. What is the total sales volume of each region and the percentage of the global sales? How has the
percentage of regional sales changed over time?

In [14]:
%sql select round(sum(na_sales), 2), round(sum(eu_sales) ,2), round(sum(jp_sales) ,2) , round(sum(other_sales) ,2) from vgsales

 * sqlite:///./Videogames.db
Done.


[(4223.21, 2370.14, 1193.52, 784.19)]

In [15]:
%%sql select round(sum(na_sales)/sum(global_sales),4)*100 as north_america, 
round(sum(eu_sales)/sum(global_sales),4)*100 as eu,
round(sum(jp_sales)/sum(global_sales),4)*100 as jp,
round(sum(other_sales)/sum(global_sales),4)*100 as other 
from vgsales 

 * sqlite:///./Videogames.db
Done.


[(49.25, 27.639999999999997, 13.919999999999998, 9.139999999999999)]

From 1980 to 2016, North America accounted for nearly half of all popular game sales, followed by
Europe and Japan, and all other regions accounted for only about 9%.
Result (in percent):
How this ratio changes over time is also a question worthy of exploration, so further inquiries,
and at the same time solve the problem of the decimal place confusion.

In [16]:
%%sql select year, 
round(cast(na/global as decimal (4,4))*100,2) as na, 
round(cast(eu/global as decimal (4,4))*100,2) as eu,
round(cast(jp/global as decimal (4,4))*100,2) as jp,
round(cast(other/global as decimal (4,4))*100,2) as other 
from (select
     year, 
     sum(na_sales) as na, 
     sum(eu_sales) as eu,
     sum(other_sales) as other,
     sum(jp_sales) as jp
     sum(global_sales) as global
     from vgsales 
     group by year) as a 
     order by year 

 * sqlite:///./Videogames.db
(sqlite3.OperationalError) near "sum": syntax error
[SQL: select year,
round(cast(na/global as decimal (4,4))*100,2) as na, 
round(cast(eu/global as decimal (4,4))*100,2) as eu,
round(cast(jp/global as decimal (4,4))*100,2) as jp,
round(cast(other/global as decimal (4,4))*100,2) as other 
from (select
     year, 
     sum(na_sales) as na, 
     sum(eu_sales) as eu,
     sum(other_sales) as other,
     sum(jp_sales) as jp
     sum(global_sales) as global
     from vgsales 
     group by year) as a 
     order by year]
(Background on this error at: https://sqlalche.me/e/20/e3q8)


It can be seen that in the years after 1980, North America accounted for about 90% of the game
market. After 1983, it declined, but remained at about 50%. After 2010, there was a relatively gentle decline.There is a 30% drop.The proportion of Japan in 1980-1982 was 0, I guess it is due to missing data. From 1983-1996 ,
Japan accounted for between 25%-50% of the global game market, and gradually declined after 1996. From
2001-2010 it was During the low tide period, it only accounts for about 10% of global sales each year.
The reason is worth exploring. 
Europe's share of sales has grown slowly and steadily, from 5% to about 40%. Other regions also
grew slowly, from 1% to 10%.


6. The games with the most sales in North America are sorted by category and platform.


In [17]:
%sql select genre, round(sum(NA_Sales),2) as sales, count(genre) as number from vgsales group by genre order by sales desc limit 10

 * sqlite:///./Videogames.db
Done.


[('Action', 830.95, 3072),
 ('Sports', 678.47, 2308),
 ('Shooter', 557.26, 1279),
 ('Platform', 421.48, 849),
 ('Misc', 404.34, 1681),
 ('Racing', 350.62, 1232),
 ('Role-Playing', 307.1, 1400),
 ('Fighting', 218.0, 832),
 ('Simulation', 175.15, 834),
 ('Puzzle', 108.81, 550)]

In [18]:
%sql select platform, round(sum(NA_Sales),2) as sales, count(genre) as number from vgsales group by platform order by sales desc limit 10

 * sqlite:///./Videogames.db
Done.


[('X360', 601.05, 1265),
 ('PS2', 583.84, 2161),
 ('Wii', 507.71, 1325),
 ('PS3', 392.26, 1329),
 ('DS', 390.44, 2162),
 ('PS', 336.51, 1196),
 ('GBA', 187.54, 822),
 ('XB', 186.69, 824),
 ('N64', 139.02, 319),
 ('GC', 133.46, 556)]

The categories with the most sales in North America are action, sports, and shooting, and the platforms
with the highest sales are X360, PS2, Wii, and PS3.

7. The games with the most sales in Europe are sorted by category and platform

In [19]:
%sql select genre, round(sum(EU_Sales),2) as sales, count(genre) as number from vgsales group by genre order by sales desc limit 10

 * sqlite:///./Videogames.db
Done.


[('Action', 510.83, 3072),
 ('Sports', 374.48, 2308),
 ('Shooter', 311.41, 1279),
 ('Racing', 233.65, 1232),
 ('Misc', 213.65, 1681),
 ('Platform', 190.56, 849),
 ('Role-Playing', 175.06, 1400),
 ('Simulation', 105.43, 834),
 ('Fighting', 99.05, 832),
 ('Adventure', 63.23, 1246)]

In [20]:
%sql select platform, round(sum(EU_Sales),2) as sales, count(genre) as numebr from vgsales group by platform order by sales desc limit 10

 * sqlite:///./Videogames.db
Done.


[('PS3', 343.71, 1329),
 ('PS2', 339.29, 2161),
 ('X360', 280.58, 1265),
 ('Wii', 268.38, 1325),
 ('PS', 213.6, 1196),
 ('DS', 194.65, 2162),
 ('PC', 139.68, 960),
 ('PS4', 123.7, 335),
 ('GBA', 75.25, 822),
 ('PSP', 68.25, 1213)]

The categories with the most sales in Europe are action, sports, and shooting, and the platforms
with the highest sales are PS3, PS2, X360, and Wii, just like the United States.

8. The games with the most sales in Japan are grouped and sorted by genre and platform.


In [21]:
%sql select genre, round(sum(JP_Sales),2) as sales, count(genre) as number from vgsales group by genre order by sales desc limit 10 

 * sqlite:///./Videogames.db
Done.


[('Role-Playing', 312.66, 1400),
 ('Action', 137.89, 3072),
 ('Sports', 133.4, 2308),
 ('Platform', 124.32, 849),
 ('Misc', 102.38, 1681),
 ('Fighting', 84.16, 832),
 ('Puzzle', 55.49, 550),
 ('Racing', 53.88, 1232),
 ('Simulation', 53.36, 834),
 ('Adventure', 49.27, 1246)]

In [22]:
%sql select platform, round(sum(JP_Sales),2) as sales,count(genre) as number from vgsales group by platform order by sales desc limit 10

 * sqlite:///./Videogames.db
Done.


[('DS', 175.57, 2162),
 ('PS', 139.82, 1196),
 ('PS2', 139.2, 2161),
 ('SNES', 116.55, 239),
 ('NES', 98.65, 98),
 ('GB', 85.12, 98),
 ('PS3', 79.99, 1329),
 ('PSP', 76.79, 1213),
 ('Wii', 69.35, 1325),
 ('GBA', 47.33, 822)]

The genres with the most sales in Japan are role-playing, action, and sports, and the platforms with the highest sales
are DS, PS, PS2, and SNES.

Final thoughts: 
The results of this type of query still needs to be presented in combination with visualisation tools, such as using bar
graphs and line graphs to display game distribution and sales that change over the years, and using pie charts to
display the percentage of total sales in each region. 